dnf5/commands/check/check.cpp

```diff
@@ -110,6 +112,62 @@ cli::ArgumentParser::NamedArg * create_option(
 }
 
 
+inline std::map<std::string, std::vector<std::string>> has_duplicate_in_PATH() {
+    std::map<std::string, std::vector<std::string>> PATH_dir;
+    std::map<std::string, std::vector<std::string>> Duplicates;
+    const std::string PATH = getenv("PATH");
+
+    // parse PATH variable
+    for (const auto path : PATH | std::views::split(':')) {
+        PATH_dir.insert({path.data(), {}});
+    }
+
+    // populate file list
+    for (const auto& [dir, _] : PATH_dir) {
+        for (const auto& entry : std::filesystem::directory_iterator(dir)) {
+            std::string path = entry.path().string();
+            std::size_t found = path.find_last_of("/\\");
```

Comment:
The backslash is wrong here as POSIX requires a forward slash. That you assume POSIX is visible few lines above when segmenting PATH environment variable by colons.